controllers/operatorpolicy_status.go

```diff
@@ -775,6 +794,13 @@ func buildDeploymentCond(
 		message = fmt.Sprintf("the deployments %s do not have their minimum availability", names)
 	}
 
+	// If the status changed, that means the condition should be NonCompliant
+	// Apply override if complianceConfig is set to Compliant
+	if status == metav1.ConditionFalse && complianceConfig == "Compliant" {
+		status = metav1.ConditionTrue
+		message += " (policy compliance is not impacted due to spec.complianceConfig.deploymentsUnavailable)"
+	}
```

Comment:
Nit: I think the `status == metav1.ConditionFalse` could be removed if this was moved into the `if len(unavailableDeps) != 0` section.

Reply:
Yea this could go either way. I was debating on this but ended up putting it outside the `if len(unavailableDeps) != 0` to avoid nested if statements. I'm leaning towards the current implementation in case in the future we decide to add more cases which would make `status` false.

Reply:
That's a good point :+1: